networkd/network_server.ml

```diff
@@ -130,13 +130,25 @@ module Sriov = struct
       else
         Ok Modprobe_successful_requires_reboot
     | None ->
-      debug "%s SR-IOV on a device: %s via sysfs" op dev;
+      (* enable: try sysfs interface to set numfvs = maxvfs. if fails, but vfs are enabled, assume manual configuration.
+         disable: Net.Sriov.disable will not be called for manually configured interfaces, as determined by `require_operation_on_pci_device` *)
+      (* let Ok(maxvfs) = Sysfs.get_sriov_maxvfs dev in *)
+      if enable then
       begin
-        if enable then Sysfs.get_sriov_maxvfs dev
-        else Sysfs.unbind_child_vfs dev >>= fun () -> Ok 0
-      end >>= fun numvfs ->
-      Sysfs.set_sriov_numvfs dev numvfs >>= fun _ ->
-      Ok Sysfs_successful
+        let present_numvfs = Sysfs.get_sriov_numvfs dev in
+        match
+          ( try Sysfs.get_sriov_maxvfs dev >>= fun (maxvfs) -> maxvfs |> Sysfs.set_sriov_numvfs dev
+          with e -> let msg = Printf.sprintf "Error: trying sysfs SR-IOV interface failed with exception %s on device: %s" (Printexc.to_string e) dev in Error(Other, msg) ),
+          present_numvfs
+        with
+        | Ok _ , _ -> debug "%s SR-IOV on a device: %s via sysfs" op dev; Ok Sysfs_successful
```

Comment:
I was a bit surprised that the code would not use `error` and `return` (as I would expect in monadic code) but the existing code does not use it either.

Reply:
I tried to stick to the original code for error handling and return types, to not mess with other code portions and keep currently expected behavior.